File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    s = set(i for i in range(1, int(yellow ** 0.5) + 1) if yellow % i == 0)


    for n in s:
        if (n+2) * 2 + yellow // n * 2 == brown:
            return [yellow // n + 2, n+2]
    

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        for (int i = 1; i < (int) Math.sqrt(yellow) + 1; i++) {
            if (yellow % i == 0 && 2 * (i+yellow / i) + 4 == brown) {
                return new int[]{yellow / i + 2 , i + 2};
            }
        }
        return new int[]{};
    }
}